### PR TITLE
Report missing models in component library

### LINKED ISSUE #128: Print a list of missing models prior to model composition
This should also be helpful for figuring out which models need to be recovered.

### PR DESCRIPTION
**Pull request by schmerl**
Resolves #128

### CODE REVIEW

File: src/rosdiscover/cli.py [hunk 1]
```diff
@@ -39,6 +39,10 @@ def launch(args) -> None:
     """Simulates the architectural effects of a `roslaunch` command."""
     summary = _launch_config(args)
     output = summary.to_dict()
+
+    # Warn of any undefined nodes
+    for node_summary in summary.unresolved:
```
Comment (ChrisTimperley) [inline src/rosdiscover/cli.py:44]: Doesn't this take place after the composition stage?

File: src/rosdiscover/interpreter/model.py [hunk 2]
```diff
@@ -28,9 +28,9 @@ def register(package: str,
 
     @staticmethod
     def find(package: str, name: str) -> 'Model':
-        try:
+        if (package, name) in Model._models.keys():
```
Comment (ChrisTimperley) [inline src/rosdiscover/interpreter/model.py:31]: there shouldn't be a need for `.keys()` here. `(package, name) in Model._models` should suffice.

File: src/rosdiscover/interpreter/summary.py [hunk 3]
```diff
@@ -90,3 +90,7 @@ def __getitem__(self, name: str) -> NodeSummary:
 
     def to_dict(self) -> List[Dict[str, Any]]:
         return [n.to_dict() for n in self.values()]
+
+    @property
+    def unresolved(self) -> Iterator[NodeSummary]:
```
Comment (ChrisTimperley) [inline src/rosdiscover/interpreter/summary.py:95]: This does also seem useful, albeit orthogonal to #128

File: src/rosdiscover/cli.py (hunk 1 above)
Comment (schmerl) [reply src/rosdiscover/cli.py:44]: I think so - but this it to do the final report of missing components (we were already logging them along the way.)

File: src/rosdiscover/interpreter/summary.py (hunk 3 above)
Comment (schmerl) [reply src/rosdiscover/interpreter/summary.py:95]: It is used in the reporting.

File: src/rosdiscover/interpreter/model.py (hunk 2 above)
Comment (schmerl) [reply src/rosdiscover/interpreter/model.py]: Done.

**Pull request closed without merging**